File: mwutil/modules/setup-github.py
import os
from argparse import ArgumentParser, Namespace

from mwutil.config import MWUtilConfig
from mwutil.exec import set_git_config
from mwutil.module import MWUtilModule


class SetupGithub(MWUtilModule):

    def get_description(self) -> str:
        return "Sets up a local repository that was cloned from Github"

    def populate_subparser(self, parser: ArgumentParser, basedir: MWUtilConfig):
        pass

    def execute(self, config: MWUtilConfig, args: Namespace):
        git_email = os.getenv("GIT_EMAIL")
        git_username = os.getenv("GIT_USERNAME")
        
        set_git_config("user.email", git_email)
        set_git_config("user.name", git_username)

        print("Done!")

File: rs/src/modules/setup_github.rs
use crate::config::MWUtilConfig;
use crate::utils::set_git_config;
use anyhow::anyhow;
use std::env;
use std::path::PathBuf;

pub fn execute(config: &MWUtilConfig, repo_folder: Option<PathBuf>) -> anyhow::Result<()> {
    let repo_folder = repo_folder.unwrap_or(env::current_dir()?);
    set_git_config(
        "user.email",
        config.git_email.clone().ok_or_else(|| anyhow!("Git email not set!"))?.as_str(),
        &repo_folder
    )?;
    set_git_config(
        "user.name",
        config.git_username.clone().ok_or_else(|| anyhow!("Git name not set!"))?.as_str(),
        &repo_folder
    )?;
    println!("Done!");
    Ok(())
}
